src/layers/VivViewerLayer/utils.js

```diff
@@ -32,24 +32,55 @@ export function isInTileBounds({
 
 export function renderSubLayers(props) {
   const {
-    bbox: { left, top, right, bottom }
+    bbox: { left, top, right, bottom },
+    x,
+    y,
+    z
+    // eslint-disable-next-line react/destructuring-assignment
   } = props.tile;
-  const { colorValues, sliderValues, tileSize, visible, opacity, data } = props;
-  const xrl =
-    // If image metadata is undefined, do not render this layer.
-    props.imageWidth &&
-    props.imageHeight &&
-    data &&
-    new XRLayer({
-      id: `XRLayer-left${left}-top${top}-right${right}-bottom${bottom}`,
-      data: null,
-      channelData: data,
-      sliderValues,
-      colorValues,
+  const {
+    colorValues,
+    sliderValues,
+    imageWidth,
+    imageHeight,
+    tileSize,
+    minZoom,
+    visible,
+    opacity,
+    data,
+    colormap,
+    dtype
+  } = props;
+  if (
+    isInTileBounds({
+      x,
+      y,
+      z: -z,
+      imageWidth,
+      imageHeight,
       tileSize,
-      bounds: [left, bottom, right, top],
-      opacity,
-      visible
-    });
-  return xrl;
+      minZoom
+    })
+  ) {
+    const xrl =
+      // If image metadata is undefined, do not render this layer.
+      imageWidth &&
+      imageHeight &&
```

Comment:
When does this happen?

Reply:
It was happening during tests due to some weirdness that neither Chuck nor I fully understand - something to do with the fact that `props` are an object with other methods so using spread syntax messes with that in our tests (which upstream deck.gl code does)